### PR TITLE
Don't split spaces

### LINKED ISSUE #70: Error when path contains a space
I think we might need to get rid of this:
https://github.com/jcansdale/gpr/blob/31072d51beeead1359447deabcaedde3686b863f/src/GprTool/IoExtensions.cs#L25-L37

...and make this a `string[]`:
https://github.com/jcansdale/gpr/blob/31072d51beeead1359447deabcaedde3686b863f/src/GprTool/Program.cs#L555

### How to test

```
gpr push 'C:\Users\Jamie Cansdale\.nuget\packages\**\*' ...
```

### PR DESCRIPTION
**Pull request by jcansdale**
Now that multiple globs can be passed in as args, we don't need to support separating globs with whitespace.

### What this PR does

- Add test for glob with space
https://github.com/jcansdale/gpr/blob/9352233faf97c6a961df726a1624c4c09ceeac3e/test/GprTool.Tests/IoExtensionsTests.cs#L16
- Remove support for separating globs with whitespace
- Don't special case `.` (treat it as a regular directory)
- Add null check for glob patterns (a package path or glob must be passed)

Fixes #70

### CODE REVIEW

**Comment by jcansdale** [comment]
@peters any chance I could get your review of this one?

How would you feel about being made a collaborator? 😄

File: test/GprTool.Tests/IoExtensionsTests.cs [hunk 1]
```diff
@@ -8,6 +8,34 @@ namespace GprTool.Tests
...
     public class IoExtensionsTests
     {
+        [TestCase("foo.nupkg", "foo.nupkg", "foo.nupkg")]
+        [TestCase("*.nupkg", "foo.nupkg;bar.nupkg", "foo.nupkg;bar.nupkg")]
+        [TestCase(".", "foo.nupkg;bar.nupkg", "foo.nupkg;bar.nupkg")]
+        [TestCase("*.nupkg", "foo bar.nupkg", "foo bar.nupkg")]
+        [TestCase("dir/foo.nupkg", "dir/foo.nupkg", "dir/foo.nupkg")]
+        [TestCase("foo bar/baz.nupkg", "foo bar/baz.nupkg", "foo bar/baz.nupkg")]
+        [TestCase("foo bar/baz.*", "foo bar/baz.nupkg", "foo bar/baz.nupkg")]
+        public void GetFilesByGlobPattern(string globPattern, string files, string expectedFiles)
```
Comment (peters) [inline test/GprTool.Tests/IoExtensionsTests.cs:18]: Need test cases for `.snupkg` and `.nupkg`, `.snupkg` is combined.

File: test/GprTool.Tests/IoExtensionsTests.cs [hunk 2]
```diff
@@ -158,33 +186,6 @@ public void GetFilesByGlobPatterns_Is_Multiple_Relative_Filenames(string relativ
             Assert.That(packages, Does.Contain(snupkgAbsoluteFilename));
         }
 
-
-        [TestCase("test.nupkg test.snupkg")]
-        [TestCase("./test.nupkg ./test.snupkg")]
-        [TestCase("./test.nupkg              ./test.snupkg", Description = "Whitespace")]
-#if PLATFORM_WINDOWS
```
Comment (peters) [inline test/GprTool.Tests/IoExtensionsTests.cs:165]: Test cases for specifically targeting Windows should not be removed?

File: test/GprTool.Tests/IoExtensionsTests.cs (hunk 1 above)
Comment (peters) [reply test/GprTool.Tests/IoExtensionsTests.cs:18]: Add test case for `globPattern` == null.

File: src/GprTool/IoExtensions.cs [hunk 3]
```diff
@@ -25,30 +25,15 @@ public static IEnumerable<string> GetFilesByGlobPattern(this string baseDirector
 
         public static IEnumerable<string> GetFilesByGlobPattern(this string baseDirectory, string globPattern, out Glob outGlob)
         {
-            var baseDirectoryGlobPattern = Path.GetFullPath(Path.Combine(baseDirectory, globPattern));
+            var baseDirectoryGlobPattern = Path.GetFullPath(Path.Combine(baseDirectory, globPattern.Trim()));
```
Comment (peters) [inline src/GprTool/IoExtensions.cs:28]: Missing null propagation check - `globPattern?.Trim() ?? string.Empty`

**Comment by peters** [comment]
> @peters any chance I could get your review of this one?
> 
> How would you feel about being made a collaborator? 😄

Thanks for the show of confidence :) I think it's best if we stay with today's model but still review each others PR's :) I think it's simplest when there is only one person responsible for maintaining a stable master :)

File: test/GprTool.Tests/IoExtensionsTests.cs (hunk 1 above)
Comment (jcansdale) [reply test/GprTool.Tests/IoExtensionsTests.cs:18]: What do you think it should do when it's null? I'm guessing it should throw?

File: src/GprTool/IoExtensions.cs (hunk 3 above)
Comment (jcansdale) [reply src/GprTool/IoExtensions.cs:28]: I'm thinking:

```cs
if (globPattern == null) throw new ArgumentNullException(nameof(globPattern));
```

Is there any reason we should accept null?

File: test/GprTool.Tests/IoExtensionsTests.cs (hunk 1 above)
Comment (peters) [reply test/GprTool.Tests/IoExtensionsTests.cs:19]: Hmm! How about `yield break` because we already properly handle an empty list of files here: https://github.com/jcansdale/gpr/blob/master/src/GprTool/Program.cs#L403

**Comment by jcansdale** [comment]
@peters,

> Thanks for the show of confidence :)

Well, your high quality PRs were appreciated!

> I think it's best if we stay with today's model but still review each others PR's :) I think it's simplest when there is only one person responsible for maintaining a stable master :)

Cool, I'll keep on tagging you then. Unfortunately I can't add you as a reviewer when you're not a collaborator. GitHub should fix that!

File: src/GprTool/IoExtensions.cs (hunk 3 above)
Comment (peters) [reply src/GprTool/IoExtensions.cs:28]: Well, the only reason I can think of is this: https://github.com/jcansdale/gpr/blob/master/src/GprTool/Program.cs#L403

**Comment by peters** [comment]
@jcansdale Ahh! I didn't know that. You can add me as a collaborator if that solves the problem 👍

File: test/GprTool.Tests/IoExtensionsTests.cs (hunk 2 above)
Comment (jcansdale) [reply test/GprTool.Tests/IoExtensionsTests.cs:165]: This was the `Multiple_Relative_Filenames` test, but multiple files are no longer supported.

File: src/GprTool/IoExtensions.cs (hunk 3 above)
Comment (jcansdale) [reply src/GprTool/IoExtensions.cs:32]: Good catch! I've added a check for this here https://github.com/jcansdale/gpr/pull/78/commits/d74b54350652281680d7a332de54192da36cd72d.

File: test/GprTool.Tests/IoExtensionsTests.cs (hunk 1 above)
Comment (jcansdale) [reply test/GprTool.Tests/IoExtensionsTests.cs:19]: I was surprised to find this is null rather than an empty list! I've added some null checks and an error message.

**Comment by jcansdale** [comment]
@peters I've added some null checks and tests since your last review. How does it look now? 🙂

**Comment by peters** [comment]
> @peters I've added some null checks and tests since your last review. How does it look now? 🙂

LGTM 😁👏
